src/ucp/api/ucp_def.h

```diff
@@ -376,18 +375,20 @@ typedef void
  * connection establishment flow.
  *
  * This structure is used for handling an incoming connection request on
- * the listener. Setting this type of handler allows to create an endpoint on
- * other worker and not limited to the worker associated with the listener.
- * Other than communication progress routines, it is allowed to call other
- * communication routines from the callback in the struct. The callback is
- * thread safe with respect to the worker it is invoked on. Also it is an
- * user's responsibility to avoid possible dead lock accessing different worker.
- * If the callback is called from different threads, this callback needs thread
- * safety support.
+ * the listener. Setting this type of handler allows creating an endpoint on
+ * any other worker and not limited to the worker on which the listener was
+ * created.
+ * @note
+ * - Other than communication progress routines, it is allowed to call
+ *   other communication routines from the callback in the struct.
+ * - The callback is thread safe with respect to the worker it is invoked on.
+ * - It is an user's responsibility to avoid possible dead lock accessing
+ *   different worker.
+ * - If the callback is called from different threads, this callback needs
```

Comment:
the callback is always called from the context of ucp_worker_progress, right? so don't understand why this note is needed..